tests/MongoDB.Driver.Core.Tests/Core/Authentication/MongoAWSAuthenticatorTests.cs

```diff
@@ -0,0 +1,188 @@
+                new IsMasterResult(new BsonDocument("ok", 1).Add("ismaster", 1)),
+                new BuildInfoResult(new BsonDocument("version", "4.0.0")));
+
+            var clientNonce = __randomByteGenerator.Generate(ClientNonceLength);
+            var serverNonce = Combine(clientNonce, __randomByteGenerator.Generate(ClientNonceLength));
+            var host = "sts.amazonaws.com";
+            var credential = new UsernamePasswordCredential("$external", "permanentuser", "FAKEFAKEFAKEFAKEFAKEfakefakefakefakefake");
+            
+            AwsSignatureVersion4.SignRequest(
+                dateTime,
+                credential.Username,
+                credential.GetInsecurePassword(),
+                null,
+                serverNonce,
+                host,
+                out var authHeader,
+                out var timestamp);
+
+            var dateTimeProviderMock = new Mock<IClock>();
+            dateTimeProviderMock.Setup(x => x.UtcNow).Returns(dateTime);
+
+            var randomByteGeneratorMock = new Mock<IRandomByteGenerator>();
+            randomByteGeneratorMock.Setup(x => x.Generate(It.IsAny<int>())).Returns(clientNonce);
+
+            var expectedClientFirstMessage = new BsonDocument()
+                .Add("r", new BsonBinaryData(clientNonce))
+                .Add("p", new BsonInt32('n'));
+            var expectedClientSecondMessage = new BsonDocument()
+                .Add("a", authHeader)
+                .Add("d", timestamp);
+
+            var serverFirstMessage = new BsonDocument()
+                .Add("s", new BsonBinaryData(serverNonce))
+                .Add("h", new BsonString(host));
+            var saslStartReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                $"{{conversationId: 1, done: false, payload: BinData(0,\"{ToBase64(ToBytes(serverFirstMessage))}\"), ok: 1}}"));
+            var saslContinueReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                "{conversationId: 1, done: true, payload: BinData(0,\"\"), ok: 1}"));
+
+            var subject = new MongoAWSAuthenticator(credential, null, randomByteGeneratorMock.Object, dateTimeProviderMock.Object);
+
+            var connection = new MockConnection(serverId);
+            connection.EnqueueReplyMessage(saslStartReply);
+            connection.EnqueueReplyMessage(saslContinueReply);
+
+            var expectedRequestId = RequestMessage.CurrentGlobalRequestId + 1;
+
+            Exception exception;
+            if (async)
+            {
+                exception = Record.Exception(() => subject.AuthenticateAsync(connection, connectionDescription, CancellationToken.None).GetAwaiter().GetResult());
+            }
+            else
+            {
+                exception = Record.Exception(() => subject.Authenticate(connection, connectionDescription, CancellationToken.None));
+            }
+            exception.Should().BeNull();
```

Comment:
There is never a need to call:

    exception.Should().BeNull();

Unless it's a test of something that sometimes throws and sometimes doesn't.

If an exception is thrown the test will fail anyway.